react/components/InputSearch/index.js

```diff
@@ -38,9 +38,17 @@ class InputSearch extends Component {
     this.props.onClear && this.props.onClear(event)
   }
 
-  handleSubmit = () => {
+  handleSubmit = event => {
     const { onSubmit, value } = this.props
-    onSubmit && onSubmit(value)
+    onSubmit &&
+      onSubmit({
+        ...event,
+        target: {
+          ...event.target,
+          value,
+        },
+        preventDefault: event.preventDefault || (() => {}),
```

Comment:
@emersonlaurentino @kevinch what do you guys think of this? Since onSubmit SyntheticEvent from react behaviour is to always return the event (which reloads the page in the normal conditions, and bubbles up the DOM), I did this to partially mimic this, so codes where you expect value to be inside `e.target` and event to have a `.preventDefault` don't break.

Reply:
I couldn't think of another solution, if you have some ideas they would be more then welcome.
I also tried creating a new event to pass up (with `new Event('submit')` or `new Event('click')` for example) but they don't have `.preventDefault` when they get to the callback 😔

Reply:
@ohmyguigs I think you can pass the event as a second part of prop onSubmit and who calls `Input Search` defines event.preventDefault.

Reply:
I just don't wanna break backward compatibility (as before the onSubmit had only the event as param)

Reply:
I approve your `gambiarra`! I couldn't imagine another way for onClick to take the input value and make it available on target.